addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -166,11 +169,30 @@ export class SuggestionService {
                 suggestions.push(partner);
             }
         }
+        specialMentionRegistry
+            .getEntries()
+            .filter(([_name, info]) => {
+                return (
+                    info.channel_types.includes(thread.channel_type) &&
+                    cleanedSearchTerm.length >= Math.min(4, info.label.toString().length) &&
+                    info.label.toString().startsWith(cleanedSearchTerm)
+                );
+            })
+            .forEach(([name, info]) =>
+                specialSuggestions.push({
+                    isSpecial: true,
+                    label: info.label,
+                    type: name,
+                    name: info.displayName,
+                    description: info.description,
+                })
+            );
```

Comment:
Usually we avoid `.forEach()`, and instead prefer `for (const x of list)`.

Reasoning:
- it's easier to spot loops in the code
- `.forEach` expression are too dense, so it forces you to make several smaller statements, which eases readability
- `forEach()` params are confusing, we could mix key and value and introduce errors
- `forEach` adds a new scope with an inner function. This is less efficient than `for of` loops. Benchmarks show it. It's usually not perceivable, but that's still an argument against `.forEach()`.